### PR TITLE
feat: Add content to the Jakarta Servlet Spec page

### PR DESCRIPTION
**Pull request by ZacharySabourin**
Resolves https://github.com/jakartaee/jakarta.ee/issues/1751

### CODE REVIEW

**Comment by ZacharySabourin** [comment]
@chrisguindon

**Comment by ZacharySabourin** [comment]
@ivargrimstad Preview page: https://deploy-preview-684--jakartaee-specifications.netlify.app/specifications/servlet/

**Comment by ivargrimstad** [comment]
Request for project team to review: https://www.eclipse.org/lists/servlet-dev/msg00627.html

File: servlet/_index.md [hunk 1]
```diff
@@ -1,7 +1,173 @@
...
+
+When the servlet URL is visited by a client, a request is sent to the host server, and then it hands off the request to the servlet container.
+The servlet container then determines which servlet to invoke and then calls upon it with the request and response objects.  
+
+The servlet life cycle is as follows:
+
+1. Loading and Instantiation: Servlet container loads the servlets that are registered and instantiates them for use.
+2. Initialization: `Servlet` initialization occurs calling upon the `init()` method of the servlet.
+3. RequestHandling: Request objects of type `ServletRequest` are handled, and then `ServletResponse` objects are filled out and passed as parameters to the `service()` method. At this stage, single requests or multiple asynchronous requests can be processed, depending upon configuration.
+4. End of Service: Once a container determines that a servlet is no longer needed, the `destroy()` method of the servlet is called to allow the servlet to release resources, if needed.
```
Comment (joakime) [inline servlet/_index.md:133]: This seems inaccurate and incredibly simplified for the Servlet spec.

Why document this here?
Isn't referencing the Servlet Spec itself more appropriate?

Off the top of my head, it looks like this ..

1. The Servlet Container loads / initializes.
2. Servlet Container is started
3. Request Handling
  a. Request received
  b. Servlet Container is selected by Context Path of Request
  c. Request is Dispatched to Servlet Container
  d. Handler Chain is determined based on url-patterns of Servlets (and optional Filters) and Dispatcher Type
  e. Handler Chain is called
  f. Eventually Servlet handles the request via the `service()` method.
  g. Response is produced
4. When Servlet Container is stopped, every Filter / Servlet / Listener that is interested in the stop can perform their stop duties
5. No more requests are dispatched to stopped Servlet Container.

File: servlet/_index.md [hunk 2]
```diff
@@ -1,7 +1,173 @@
...
+`Servlets` can access information pertaining to attributes and headers via provided APIs. Attributes are associated with each request, and they can be set to specify information.
+Headers are also associated with every HTTP request, and they can be used to specify information about sessions and caching, amongst other things.
+
+`Servlets` can make use of the URI path to invoke different resources. The path portion of a URI identifies which resource should be processed.
+Part of the `Jakarta Servlet` specification outlines a standard set of guidelines that should be followed by compliant containers when processing URI paths.
+A servlet request path conforms to the following format:
+
+```Java
+requestURI = contextPath + servletPath + pathInfo
+```
```
Comment (joakime) [inline servlet/_index.md:164]: Reconstructing a Request URI is 

* scheme (from combination of configuration, connectors, and request details)
* host / authority (from combination of configuration, connectors, request details, forwarding headers, etc)
* context path (with path params merged from original submitted request path)
* servlet path (with path params merged from original submitted request path)
* path info (optional)
* query (optional)

This get more complicated when RequestDispatcher is used (with regards to merging of original request details and request dispatcher details)

File: servlet/_index.md [hunk 3]
```diff
@@ -1,7 +1,173 @@
...
+
+`Jakarta Servlet` containers utilize non-blocking request processing, which helps to improve scalability and makes it easy to increase the number of simultaneously processed connections.
+Non-Blocking IO makes it possible to read and write when needed, without requiring a connection to wait for other processes to complete. `Jakarta Servlet` provides an API for working with non-blocking requests.
+
+`HTTP/2 Server Push` enables a servlet to push required resources to a client when needed. This provides the ability to load resources before they are needed so that overall performance of servlet applications is significantly improved.
```
Comment (joakime) [inline servlet/_index.md:169]: This is completely deprecated.
The IETF has deprecated HTTP/2 Server Push.
The browsers have deprecated HTTP/2 Server Push.
Even HTTP/3 has no Server Push anymore.

**Review by joakime — CHANGES_REQUESTED** [review]
Move the non-Servlet spec mentions out to their own Spec documentation.

Perhaps this text should be simplified even more and link to the spec documentation for details on things like the life cycle or how to get the request details.
Just show a simple HttpServlet code and mention that things like Filters exist, and the lifecycle and request URI details are documented in the spec documentation.

I noticed that the most common question on the servlet spec isn't even mentioned here.
How to serve static files.  This is complicated enough by the DefaultServlet and how url-patterns are resolved, so I would just mention that this is possible and then link to the spec documentation.

File: servlet/_index.md (hunk 2 above)
Comment (joakime) [reply servlet/_index.md:164]: This description also doesn't handle the nuance of how uri encoding / uri decoding / uri canonicalization impacts the resulting requestURI.
This is also a section that's best handled by pointing to the spec documentation.

File: servlet/_index.md [hunk 4]
```diff
@@ -1,7 +1,173 @@
...
+{{< grid/div class="specifications-content" >}}
+
+## Jakarta Servlet Explained
+
+The [Jakarta EE Platform](/specifications/platform) offers a number of APIs that can be used to create web based user interfaces. The most mature and perhaps most well known is the `Jakarta Servlet` specification.
+`Jakarta Servlet` was first released in 1996, and it enables Java developers to create web pages containing dynamic web content using pure Java.
+The specification has changed much over the years and it is able to interact with modern technologies and comply with the modern era of web technologies with features such as HTTP/2.
+
+In general, `Jakarta Servlet` APIs are the least often used for development of application user interfaces since the learning curve is often a bit higher.
+Although it may not be as widely used for development, it does contain a standard API which is widely supported, and most other user interface specifications are built upon it.
```
[the remark this answers is not in the document]
Comment (markt-asf) [reply servlet/_index.md:20]: +1 to @joakime's comments and -1 to the above text being added.

File: servlet/_index.md [hunk 5]
```diff
@@ -1,7 +1,173 @@
...
+
+Jakarta Faces Dependency:
+
+```xml
+<dependency>
+    <groupId>jakarta.faces</groupId>
+    <artifactId>jakarta.faces-api</artifactId>
+    <version>4.0.1</version>
+</dependency>
+```
```
[the remark this answers is not in the document]
Comment (markt-asf) [reply servlet/_index.md:58]: +1 - this is for too detailed for the index page and the content for the other specs does not belong here.

File: servlet/_index.md [hunk 6]
```diff
@@ -1,7 +1,173 @@
...
+<servlet>
+    <servlet-name>Faces Servlet</servlet-name>
+    <servlet-class>jakarta.faces.webapp.FacesServlet</servlet-class>
+    <load-on-startup>1</load-on-startup>
+</servlet>
+<servlet-mapping>
+    <servlet-name>Faces Servlet</servlet-name>
+    <url-pattern>*.xhtml</url-pattern>
+</servlet-mapping>
+```
```
[the remark this answers is not in the document]
Comment (markt-asf) [reply servlet/_index.md:80]: +1 - this does not belong anywhere in the Servlet pages let alone on the index page.

File: servlet/_index.md (hunk 2 above)
Comment (markt-asf) [reply servlet/_index.md:164]: Yet another +1 and I'll add SNI and virtual hosting to the list of things that can impact mapping.

File: servlet/_index.md [hunk 7]
```diff
@@ -1,7 +1,173 @@
...
+`Servlets` can make use of the URI path to invoke different resources. The path portion of a URI identifies which resource should be processed.
+Part of the `Jakarta Servlet` specification outlines a standard set of guidelines that should be followed by compliant containers when processing URI paths.
+A servlet request path conforms to the following format:
+
+```Java
+requestURI = contextPath + servletPath + pathInfo
+```
+
+`Jakarta Servlet` containers utilize non-blocking request processing, which helps to improve scalability and makes it easy to increase the number of simultaneously processed connections.
+Non-Blocking IO makes it possible to read and write when needed, without requiring a connection to wait for other processes to complete. `Jakarta Servlet` provides an API for working with non-blocking requests.
```
Comment (markt-asf) [inline servlet/_index.md:167]: The Servlet spec should not be making statements about how containers implement it.

File: servlet/_index.md (hunk 3 above)
Comment (markt-asf) [reply servlet/_index.md:169]: And the current Servlet spec is making it optional with it likely not implemented by most containers.

File: servlet/_index.md [hunk 8]
```diff
@@ -1,7 +1,173 @@
 ---
 title: "Jakarta Servlet"
 summary: "Jakarta Servlet defines a server-side API for handling HTTP requests and responses."
+description: "Find all releases of the Jakarta Servlet specification which offers a number of APIs that can be used to create web based user interfaces."
```
Comment (markt-asf) [inline servlet/_index.md:4]: It isn't a number of AIs, it is one API - the Servlet API.

Why the focus on the user interface? The Servlet API supports more than just UI.

**Review by markt-asf — CHANGES_REQUESTED** [review]

**Comment by pmd1nh** [comment]
https://deploy-preview-684--jakartaee-specifications.netlify.app/specifications/servlet/6.1/

```
Minimum Java SE Version

Java SE 11 or higher
```

Should it be Java 17 or higher?

**Comment by markt-asf** [comment]
Nope. At this point Servlet requires Java 11 minimum.

**Pull request closed without merging**